internal/p2p/peermanager.go

```diff
@@ -975,6 +979,8 @@ func (m *PeerManager) Advertise(peerID types.NodeID, limit uint16) []NodeAddress
 		}
 	}
 
+	meanAbsScore := (totalAbsScore + 1) / (len(scores) + 1)
```

Comment:
total absolute score (and maybe arguably total score too) made the chance of hitting the coin flip lower than maybe it should have been, and this evens things up a bit more. I've multipled the meanAbsScore by 2, but we could also add/multiply the score of the peer in question. 

I guess there's a lot of "math" here to make sure that we actually have higher scored peers represented more frequently in the pex messages than lower ranked peers, and it's got to be at least a little random. I'm willing to do whatever with these numbers to get something kinda like that, but we're definitely (in this part of the code) very deep into "finger painting with numbers" territory

Reply:
Hmm. If we don't care whether scores are positive or negative, do you think we could ignore score for the purposes of choosing a selection probability, and use only the number of peers under consideration? (After all, they're already weighted by score anyway, in terms of where they come up in the list)

I should say: I don't have any objection to the overall goal here, I'm just having trouble figuring out how I'd explain what it's doing to someone else. 🙂

Reply:
That's fair, I'd like to merge this because it's an evolution of what we currently have, and fixes a potential panic, and I think preserves the general "higher ranked peers are more likely to show up in pex packet" expectation.

Reply:
It seems like a reasonable experiment, and I don't think it's _unsafe_ in any clear way, so 👍🏼